### DISCUSSION TITLE
How to upgrade nodepool instance type?

### DISCUSSION THREAD

**Discussion by Thijmen** [vitobotta/hetzner-k3s#506] [Q&A]
Hi there,

Say I want to upgrade the nodepool instance type, how would I do this? Since updating the instance type does not work it seems, my best guess is creating a new nodepool, drain and cordon the old nodes and remove the old one?

Just wanted to know the "best way" to do this, thanks :)

**Comment by vitobotta** [comment]
At the moment yes, the best way is to just replace the node pool with another the way you guessed. In the future it will be a bit easier.

**Reply by Thijmen** [reply]
Alright, thanks! And how would this apply for master nodes?

**Reply by vitobotta** [reply]
It’s a bit tricky with the master nodes right now, so it’s best to wait for the next release. It'll have changes that make these transitions smoother and easier :)

**Reply by Thijmen** [reply]
But will that help with existing clusters as well?

**Reply by vitobotta** [reply]
Yep! I need to finish something and more testing but yes, it will make things easier even with clusters created with v1 for a smooth upgrade.

**Reply by abhishek-ch** [reply]
Is this feature now available with v2.1.0?

**Reply by vitobotta** [reply]
@abhishek-ch Yes but I haven't written a doc page for it yet

**Reply by abhishek-ch** [reply]
Any expected date for the doc update ?

**Comment by vitobotta** [comment]
@Thijmen @abhishek-ch See https://github.com/vitobotta/hetzner-k3s/blob/main/docs/Upgrading_a_cluster_from_1x_to_2x.md
